fdbclient/ClientLogEvents.h

```diff
@@ -76,18 +131,76 @@ namespace FdbClientLogEvents {
 				return serializer(ar, latency, valueSize, key);
 		}
 
+		int readId;
 		double latency;
 		int valueSize;
 		Key key;
```

Comment:
In general, it will be nice to only have fields inside these Events that will be common to all the different consumers like trace log, database log, request stats. StorageContacted and readId are not being serialized. StorageContacted can be added to serialize method. If readId is only specific to request stats, then it can be passed in as a parameter to addToRequestStats and not be stored within the Events.

Reply:
Could you please explain the use of readId? Since there is no test for RequestStats, it is not apparent how useful readId is.

Reply:
The use of readId is to identify the situation where a single range read is processed by multiple storage servers because it crosses shard boundaries. This is a common situation for our workloads at Snowflake.